src/orchestrator/orchestrator.rs

```diff
@@ -265,12 +267,26 @@ impl Borrow<ArtifactPath> for ProducedArtifact {
 
 impl<'a> Orchestrator<'a> {
     pub async fn run(self, output: &mut Vec<ArtifactPath>) -> Result<HashMap<Uuid, Error>> {
-        let (results, errors) = self.run_tree().await?;
+        let token = CancellationToken::new();
+        let cloned_token = token.clone();
```

Comment:
I'd just clone it where needed instead - why use a variable here?

Reply:
I was following the examples on the tokio.rs site (https://tokio.rs/tokio/topics/shutdown)
We can also just
`let (results, errors) = self.run_tree(token.clone()).await?;`

Reply:
Well do to the async move in 273 we need to clone the token beforehand. But I renamed the variables

Reply:
Ah, my bad, you're right - we need the variable here so that we can move it into the async closure.